services/121-service/src/registration/registrations.service.ts

```diff
@@ -556,43 +576,70 @@ export class RegistrationsService {
         registration.referenceId,
       );
     if (calculatedRegistration) {
-      return this.getRegistrationFromReferenceId(
-        calculatedRegistration.referenceId,
-      );
+      return this.getRegistrationFromReferenceId({
+        referenceId: calculatedRegistration.referenceId,
+      });
     }
 
     if (process.env.SYNC_WITH_THIRD_PARTIES) {
-      await this.syncUpdatesWithThirdParties(registration, [attribute]);
+      await this.sendContactInformationToIntersolve(registration);
     }
 
-    return this.getRegistrationFromReferenceId(savedRegistration.referenceId, [
-      'program',
-    ]);
+    return this.getRegistrationFromReferenceId({
+      referenceId: savedRegistration.referenceId,
+      relations: ['program'],
+    });
   }
 
-  private async syncUpdatesWithThirdParties(
+  private async sendContactInformationToIntersolve(
```

Comment:
I had an idea about this.

Would it make sense to call this endpoint always when `reissuing a new card` than:
-  this logic does not have live in the registrationService 
- we can get rid of the ambiguous SYNC_WITH_THIRD_PARTIES env
- this function is not called for non visa fsp 
As far as I can now think of from the top of my head this is the only time that it is relevant that this info is updated. Or is it still possible to update this info after the card has been send and the person lost/changed his phone while the card is being send

Reply:
Sounds very good to consider, but I cannot oversee the full consequences of this in 10 seconds :)

Reply:
Ask intersolve